doc/onboarding-storage-provisioners.md

```diff
@@ -0,0 +1,21 @@
+# Onboarding a storage provider to CDI
+
+## Introduction
+
+As explained extensively in the [storage profile](./storageprofile.md) documentation, CDI keeps internal knowledge  
+about a wide range of storage providers in order to provide defaults & hints as they are needed.  
+This doc is about "educating" CDI about a new storage provider.
+
+## StorageProfile.Spec fields
+In order to onboard a storage provider, one should be able to pick the correct values for the storage profile [parameters](./storageprofile.md#parameters).
+
+For example, for ceph rbd, CDI knows that:
+- These are the access/volume modes (in this preference order) supported
+https://github.com/kubevirt/containerized-data-importer/blob/c6089cbcb01ab58e75b25be1770ce0f3f4625014/pkg/storagecapabilities/storagecapabilities.go#L255-L261
+- The preferred clone strategy is csi-clone (internally creates a temporary snapshot)
+https://github.com/kubevirt/containerized-data-importer/blob/c6089cbcb01ab58e75b25be1770ce0f3f4625014/pkg/storagecapabilities/storagecapabilities.go#L115-L124
+- Ceph rbd scales better when golden sources are stored as snapshots (more info in [clone-from-volumesnapshot-source](./clone-from-volumesnapshot-source.md))
+https://github.com/kubevirt/containerized-data-importer/blob/c6089cbcb01ab58e75b25be1770ce0f3f4625014/pkg/storagecapabilities/storagecapabilities.go#L109-L112
+
```

Comment:
Maybe explain the steps a storage vendor needs to consider when adding defaults to the storage profile file:
1. In general the preferred order of volume/access modes is RWX Block/RWX filesystem/RWO Block/RWO filesystem. So they need to ensure their defaults are in that order for whatever combinations are supported. Maybe explain why block is preferred (fewer layers and thus better performance), and why RWX is preferred (Live Migration)
2. Same for cloning strategy, I would say csi-clone is preferred in general, since that gives the provider the most flexibility to internally optimize the clone. Snapshot is for the cases csi clone is not supported. And also explain if neither is set it defaults to host assisted clone.
3. The golden images flow preference comes from that in certain providers creating a snapshot is many times more expensive than restoring a snapshot. Maybe just mention this so that the providers can decide for themselves which one is better. Also snapshots are entirely immutable which is also a nice property when considering golden images.

Reply:
done, let me know what you think